src/Components/Header/Header.js

```diff
@@ -67,6 +64,21 @@ const Header = ({ handleTextfieldChange }) => {
     }
   };
 
+  const createMenuItems = (optionList) => {
+    const menuItemKeys = Object.keys(optionList);
+    const menuItemLabels = Object.values(optionList);
```

Comment:
`Object.entries(obj)` returns `[key, value][]`. I think it would be better to loop through that instead of looping through one Array, and following with the index of the other.